includes/admin/services/class-rop-facebook-service.php

```diff
@@ -577,36 +577,166 @@ private function prepare_for_sharing( $post_details ) {
-
-			return false;
-		} catch ( Facebook\Exceptions\FacebookSDKException $e ) {
-			$this->logger->alert_error( 'Unable to share post for facebook.  Error: ' . $e->getMessage() );
-			$this->rop_get_error_docs( $e->getMessage() );
-
-			return false;
-		}
-	}
+    private function try_post( $new_post, $page_id, $token, $post_id, $posting_type ) {
+        $path = '/' . $page_id . '/feed';
+        switch ( $posting_type ) {
+            case 'photo':
+                $path = '/' . $page_id . '/photos';
+                break;
+            case 'video':
+                $path = '/' . $page_id . '/videos';
+                break;
+            default:
+                break;
+        }
+
+        $this->set_api( $this->credentials['app_id'], $this->credentials['secret'] );
+
+        if ( $this->get_api() ) {
+            // page was added using user application (old method)
+            $api = $this->get_api();
+            try {
+                $api->post( $path, $new_post, $token );
+
+                return true;
+            } catch ( Facebook\Exceptions\FacebookResponseException $e ) {
+                $errorMsg = $e->getMessage();
+
+                if ( strpos($errorMsg, '(#100)' ) !== false && !empty( $new_post['name'] )) {
+                    // https://developers.facebook.com/docs/graph-api/reference/v3.2/page/feed#custom-image
+                    // retry without name
+                    unset( $new_post['name'] );
+
+                    try {
+                        $api->post( $path, $new_post, $token );
+                        return true;
+                    } catch ( Facebook\Exceptions\FacebookResponseException $e ) {
+                        $this->logger->alert_error('Unable to share post for facebook. (FacebookResponseException) Error: ' . $e->getMessage());
+
+                        return false;
+                    } catch ( Facebook\Exceptions\FacebookSDKException $e ) {
+                        $this->logger->alert_error('Unable to share post for facebook.  Error: ' . $e->getMessage());
+
+                        return false;
+                    }
+
+                } else {
+                    $this->logger->alert_error( 'Unable to share post for facebook. (FacebookResponseException) Error: ' . $errorMsg);
+
+                    return false;
+                }
+            } catch ( Facebook\Exceptions\FacebookSDKException $e ) {
+                $this->logger->alert_error( 'Unable to share post for facebook.  Error: ' . $e->getMessage() );
+
+                return false;
+            }
+        } else {
+            // page was added using ROP application (new method)
+
+            $post_data = $new_post;
+            $post_data['access_token'] = $token;
+
+            if ( $posting_type === 'video' ) {
+                $url = 'https://graph-video.facebook.com' . $path;
+            } else {
+                $url = 'https://graph.facebook.com' . $path;
+            }
+
+            $ch = curl_init();
```

Comment:
Please try accomplishing this using Guzzle which is already integrated in the plugin